arbnode/transaction_streamer.go

```diff
@@ -1568,7 +1569,12 @@ func (s *TransactionStreamer) submitEspressoTransactions(ctx context.Context) er
 			return fmt.Errorf("failed to build the hotshot transaction: a large message has exceeded the size limit or failed to get a message from storage")
 		}
 
-		payload, err = arbutil.SignHotShotPayload(payload, s.getAttestationQuote)
+		signFunc := s.getAttestationQuote
+		if s.EspressoKeyManager != nil {
+			signFunc = s.EspressoKeyManager.Sign
```

Comment:
The sign function of EspressoKeyManager should do normal ECDSA signing, it shouldnt generate attestation quote

Reply:
This is just determining if we are using the key manager and if we are using the signing func that it has to sign the hotshot payload.

I thing jeremy did this because the PR is targeting the integration branch with this change.

I think we should just be targeting the feature branches with this work so we don't need to worry about breaking the feature set of our release branches until this feature work is done. I would agree that we should make the poster just sign the batch with the ecdsa key for the sake of this feature.

Reply:
Yes that makes sense!

@ImJeremyHe  we should merge this code with `celestia-feat-stateless-batcher`